cmd/kubeadm/app/cmd/init.go

```diff
@@ -138,14 +142,15 @@ func NewCmdInit(out io.Writer) *cobra.Command {
 		"The discovery method kubeadm will use for connecting nodes to the master",
 	)
 
-	return cmd
-}
+	cmd.PersistentFlags().StringVar(
+		&deploymentType, "deployment", deploymentType,
```

Comment:
@dgoodwin Please add a `ControlPlane` struct to the `MasterConfiguration` API. 
In that struct, add this `DeploymentType string` field.

Also, make a custom flag for this like I and @mikedanese have done earlier